q2cli/tests/test_tools.py

```diff
@@ -130,6 +135,64 @@ def test_merged_metadata(self):
         self.assertIn("IDS:  1", result.output)  # only 1 ID is shared
         self.assertIn("COLUMNS:  4", result.output)
 
+    def test_export_to_dir_w_format(self):
+        output_path = os.path.join(self.tempdir, 'output')
+        result = self.runner.invoke(tools, [
+            'export', '--input-path', self.ints1, '--output-path', output_path,
+            '--output-format', 'IntSequenceDirectoryFormat'
+        ])
+        self.assertEqual(result.exit_code, 0)
+        self.assertTrue(os.path.isdir(output_path))
+
+    def test_export_to_dir_no_format(self):
+        output_path = os.path.join(self.tempdir, 'output')
+        self.runner.invoke(tools, [
+            'export', '--input-path', self.viz, '--output-path', output_path
+        ])
+
+        self.assertTrue(os.path.isdir(output_path))
+        self.assertIn('index.html', os.listdir(output_path))
+        self.assertIn('index.html', os.listdir(output_path))
+
+    def test_export_to_file(self):
+        output_path = os.path.join(self.tempdir, 'output')
+        os.mkdir(output_path)
+        self.runner.invoke(tools, [
+            'export', '--input-path', self.ints1, '--output-path', output_path,
+            '--output-format', 'IntSequence1'
+            ])
+        self.assertIn('ints1.qza', os.listdir(output_path))
+
+    def test_export_visualization_to_dir(self):
+        output_path = os.path.join(self.tempdir, 'output/viz.qza')
+        self.runner.invoke(tools, [
+            'export', '--input-path', self.viz, '--output-path', output_path
+        ])
+
+        self.assertIn('index.html', os.listdir(output_path))
+        self.assertIn('index.tsv', os.listdir(output_path))
+        self.assertTrue(os.path.isdir(output_path))
+
+    def test_export_visualization_w_format(self):
+        output_path = os.path.join(self.tempdir, 'output')
+        result = self.runner.invoke(tools, [
+            'export', '--input-path', self.viz, '--output-path', output_path,
+            '--output-format', 'IntSequenceDirectoryFormat'
+        ])
+        self.assertEqual(result.exit_code, 1)
+        self.assertIn('visualization', result.output)
+        self.assertIn('--output-format', result.output)
+
+    def test_export_path_file_is_replaced(self):
+        output_path = os.path.join(self.tempdir, 'helloWorld')
+        os.mkdir(output_path)
+
+        result = self.runner.invoke(tools, [
+            'export', '--input-path', self.ints1, '--output-path', output_path,
+            '--output-format', 'IntSequence1'
+        ])
+        self.assertNotIn('helloWorld', result.output)
+
```

Comment:
I was going to comment that we're missing a test to see if a directory is replaced, but in retrospect, that probably doesn't make a lot of sense. No reason to delete the directory before-hand, that could cause a lot of unhappy users.